src/main/scala/nodes/learning/KernelRidgeRegression.scala

```diff
@@ -0,0 +1,343 @@
+package nodes.learning
+
+import workflow.LabelEstimator
+
+import breeze.linalg._
+import breeze.math._
+import breeze.numerics._
+import org.apache.spark.rdd.RDD
+import org.apache.spark.broadcast.Broadcast
+import org.apache.spark.SparkConf
+import org.apache.spark.SparkContext
+import org.apache.spark.SparkContext._
+import utils._
+
+
+import edu.berkeley.cs.amplab.mlmatrix.RowPartitionedMatrix
+import edu.berkeley.cs.amplab.mlmatrix.util.{Utils => MLMatrixUtils}
+
+import scala.util.Random
+
+class KernelRidgeRegression(
+  kernelBlockGenerator: KernelGenerator,
```

Comment:
Morally/mathematically its the same thing since if you minimize ||Ax - b|| for a PD A, you will recover A^{-1}b. Just implementation wise, since we know the kernel is PD we want to do the linear system solve. Am I right @shivaram?